include/drivers/ipm.h

```diff
@@ -221,6 +346,185 @@ static inline int z_impl_ipm_set_enabled(const struct device *ipmdev,
 	return api->set_enabled(ipmdev, enable);
 }
 
+#ifdef CONFIG_IPM_MULTICHANNEL_DT_SUPPORT
+
+/**
+ * @brief Try to send a message over the IPM device through a specific channel.
+ *
+ * If the hardware supports it, the message is sent through a specific channel.
+ *
+ * @param ipmdev Driver instance
+ * @param channel IPM device channel
+ * @param wait If nonzero, busy-wait for remote to consume the message. The
+ *	       message is considered consumed once the remote interrupt handler
+ *	       finishes. If there is deferred processing on the remote side,
+ *	       or you would like to queue outgoing messages and wait on an
+ *	       event/semaphore, you can implement that in a high-level driver
+ * @param id Message identifier. Values are constrained by
+ *        @a ipm_max_data_size_get since many boards only allow for a
+ *        subset of bits in a 32-bit register to store the ID.
+ * @param data Pointer to the data sent in the message.
+ * @param size Size of the data.
+ *
+ * @retval -EBUSY    If the remote hasn't yet read the last data sent.
+ * @retval -EMSGSIZE If the supplied data size is unsupported by the driver.
+ * @retval -EINVAL   If there was a bad parameter, such as: too-large id value.
+ *                   or the device isn't an outbound IPM channel.
+ * @retval 0         On success.
+ */
+__syscall int ipm_send_channel(const struct device *ipmdev, ipm_channel_id_t channel,
+			       int wait, uint32_t id, const void *data, int size);
+
+static inline int z_impl_ipm_send_channel(const struct device *ipmdev, ipm_channel_id_t channel,
+					  int wait, uint32_t id, const void *data, int size)
+{
+	const struct ipm_driver_api *api =
+		(const struct ipm_driver_api *)ipmdev->api;
+
+	if (api->send_channel == NULL) {
+		return -ENOSYS;
+	}
+
+	return api->send_channel(ipmdev, channel, wait, id, data, size);
+}
+
+/**
+ * @brief Enable interrupts and callbacks for inbound channels.
+ *
+ * Enable interrupts or setup the device for the specified inbound channels.
+ *
+ * @param ipmdev Driver instance pointer.
+ * @param channel IPM device channel
+ * @param enable Set to 0 to disable and to nonzero to enable.
+ *
+ * @retval 0       On success.
+ * @retval -EINVAL If it isn't an inbound channel.
+ */
+__syscall int ipm_set_enabled_channel(const struct device *ipmdev, ipm_channel_id_t channel,
+				      int enable);
+
+static inline int z_impl_ipm_set_enabled_channel(const struct device *ipmdev,
+						 ipm_channel_id_t channel, int enable)
+{
+	const struct ipm_driver_api *api =
+		(const struct ipm_driver_api *)ipmdev->api;
+
+	if (api->set_enabled_channel == NULL) {
+		return -ENOSYS;
+	}
+
+	return api->set_enabled_channel(ipmdev, channel, enable);
+}
+
+/**
+ * @brief Return the maximum number of channels.
+ *
+ * Return the maximum number of channels supported by the hardware.
+ *
+ * @param ipmdev Driver instance pointer.
+ *
+ * @return Maximum possible number of supported channels.
```

Comment:
I'm thinking if `IPM_MULTICHANNEL_DT_SUPPORT` is necessary. This function could return 1 if given hardware does not support multiple channels.

Reply:
Oh you are right, I'll try to implement that after having gathered more comments about this PR.